pkg/migration/controllers/migration.go

```diff
@@ -1013,6 +1021,15 @@ func (m *MigrationController) checkAndUpdateDefaultSA(
 			destSA.ImagePullSecrets = append(destSA.ImagePullSecrets, s)
 		}
 	}
+	// merge annotation for SA
+	if destSA.Annotations != nil {
```

Comment:
This should be switched. You should be checking for sourceSA.Annotation != nil and create a destSA.Annotations map.

Reply:
This is migration path - 
- we try to create service account on destination cluster
if already exists- 
  - fetch destination service account
  - update destination annotation with source cluster annotation (merge in case of same key found)